### PR TITLE
22 create a decorator for restricting amount of time a function can be called

### PR DESCRIPTION
**Pull request by Dawid64**
Added requested decorator ending #22

### CODE REVIEW

**Review by adamtms — APPROVED** [review]
Looks good.
But, I have an idea for an optional improvement.
In most cases number of API calls is tracked using tumbling windows not slididing windows and these windows are the same for all users. e.g. if you can send 100 request per hours, you could send 100 requests between 5:50-5:59 and then additional 100 between 6:01-6:10.
We can always add this as a feature later.

**Comment by Dawid64** [comment]
> you could send 100 requests between 5:50-5:59 and then additional 100 between 6:01-6:10.
I applied with 2 versions, syncing to clock, to do mentioned functionality, and slightly fast - creating start of interval at the beginning of function

File: decorify/basic.py [hunk 1]
```diff
@@ -106,3 +107,152 @@ def wrapped(*args, **kwargs):
...
+    Processing 1
+    >>> example_function(2)  # Immediate execution
+    Processing 2
+    >>> example_function(3)  # Waits until time limit allows
+    Processing 3
+    """
+    __func__.__interval_counter = 0
+    if sync_with_clock:
+        now = datetime.datetime.now()
+        __func__.__interval_number = (now.hour * 360 + now.minute * 60 + now.second) // time
```
Comment (adamtms) [inline decorify/basic.py:223]: Shouldnt it be 3600 instead of 360?

File: decorify/basic.py
Comment (adamtms) [inline decorify/basic.py:1]: I am also wondering whether we want to look at the clock or just find interval number from timestamp, current solution can kind of break if we pass e.g. 24h. However it is and edge case I dont know why would anyone use this function then

**Review by adamtms — CHANGES_REQUESTED** [review]

File: decorify/basic.py
Comment (Dawid64) [reply decorify/basic.py:1]: I will add time, and I will leave a note not to use it for over 14 days

Comment (Dawid64) [reply decorify/basic.py:1]: I found even better and simpler way for that, now it should not only be much quicker, but it also removes upper bound.

Comment (adamtms) [reply decorify/basic.py:1]: yea it looks much simpler.

**Review by JaszczurGra — APPROVED** [review]
Not tested

**Pull request merged**